### DISCUSSION TITLE
Overloaded effect operations?

### DISCUSSION THREAD

**Discussion by chtenb** [koka-lang/koka#588] [Q&A]
Example:

```koka
pub effect sample<a>
  ctl uniform/add-case(value : a) : ()
  ctl weighted/add-case(weight : int, value : a) : ()
```

This won't compile: 

```
repro.kk(3,23): parse error: invalid syntax
  qualified identifier
```

However, this won't either:

```koka
pub effect sample<a>
  ctl add-case(value : a) : ()
  ctl add-case(weight : int, value : a) : ()
```

```
repro.kk(4, 7): type error: definition @select-add-case is already defined in this module, at (3, 7)
  hint: use a local qualifier?
```

Is this an inherent limitation of effect operations?

**Comment by TimWhiting** [comment]
I don't think this is a necessary limitation, but you could probably achieve the same effect with.

```
pub type sample-value<a>
  Uniform(value: a)
  Weighted(weight: int, value: a)

pub effect sample<a>
  ctl add-case(value : sample-value<a>) : ()
```

```
add-case(Uniform(10))
add-case(Weighted(10, 10))

// versus

uniform/add-case(10)
weighted/add-case(10, 10)
```

Though, I guess static overloading could choose the right one based off of number of parameters in this case, so the overloaded version would be slightly less verbose, although it could be confusing to readers to see a function called with different numbers of arguments without any clarification on what the difference is - but we have that with optional parameters anyways.

@daanx Is there a reason we limited qualifiers in effect operations?
